### PR TITLE
#2021 add HEAD support - minimal

### PR DESCRIPTION
**Pull request by jocutajar**
Could it be as simple as this?

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -285,7 +285,8 @@ def test_do_GET(self):
 
     def test_do_OPTIONS(self):
         self.assertIsNotNone(MockRestApiServer(RestApiHandler, 'OPTIONS / HTTP/1.0'))
-
+    def test_do_HEAD(self): 
+         self.assertIsNotNone(MockRestApiServer(RestApiHandler, 'HEAD / HTTP/1.0'))
```
Comment (CyberDem0n) [inline tests/test_api.py:289]:
```suggestion

    def test_do_HEAD(self): 
         self.assertIsNotNone(MockRestApiServer(RestApiHandler, 'HEAD / HTTP/1.0')) 

```

**Comment by jocutajar** [comment]
Ok, added a unit test and adjusted the white space. Thanks for the swift review @CyberDem0n !

**Comment by jocutajar** [comment]
Oh dear, does it fail on the trailing white space or something else? It's not quite clear to me.

**Comment by CyberDem0n** [comment]
> Oh dear, does it fail on the trailing white space or something else?

Yes: https://github.com/zalando/patroni/runs/7339310338?check_suite_focus=true#step:5:9
```
running flake8
tests/test_api.py:289:28: W291 trailing whitespace
tests/test_api.py:290:83: W291 trailing whitespace
```

**Comment by jocutajar** [comment]
There's also the warning on crypto - not related to this PR I suppose:

> /opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/pysyncobj/encryptor.py:3
>   /opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/pysyncobj/encryptor.py:3: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
>     import cryptography

I'll fix the white space one more time! :D

**Comment by CyberDem0n** [comment]
> warning on crypto

yes, it is unrelated.


Please also update docs: https://github.com/zalando/patroni/blob/master/docs/rest_api.rst#health-check-endpoints

**Comment by jocutajar** [comment]
> Please also update docs

Done, also did the haproxy.tmpl

**Comment by jocutajar** [comment]
Oh, what's the process now? Shall I do something?

**Comment by hughcapet** [comment]
👍

File: patroni/api.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def _write_status_code_only(self, status_code):
         self.log_request(status_code)
 
     def _write_response(self, status_code, body, content_type='text/html', headers=None):
+        # TODO: try-catch ConnectionResetError: [Errno 104] Connection reset by peer and log it in DEBUG level
```
Comment (hughcapet) [inline patroni/api.py:41]: Why do you implement neither of the options [you mentioned in the issue](https://github.com/zalando/patroni/issues/2021#issuecomment-1184118414)? Was there any problem?

Comment (jocutajar) [reply patroni/api.py:41]: Hi @hughcapet , peace by peace I suppose. incremental improvement, continuous integration :) The support for HEAD seamed like a trivial change good on it's own. Catching and logging the exceptions or sending extra headers will need some actual development, like a number of unit tests, integration tests, and more rigor in assessing the impact.

**Comment by jocutajar** [comment]
> wait a little bit

What bit sizes do you have? After the swift work with @CyberDem0n it's just sitting here. Frankly, I'm fairly disappointed with investing my time in corp-opensource. This is not the first occasion. Well, not with Zalando, alright. It's not the end of the day either. I'll move on and think twice next time. Imagine I spent more time on this. Sound judgment to start with minimal change. Gosh, rant is over. Sorry to bother you.

**Comment by hughcapet** [comment]
Oh, please calm down. If you return `OPTIONS ` to the docs (as you don't remove the support entirely), this PR can be merged.
But keep in mind, it will not trigger a new patroni release.

Btw, like you approach of participating in open source projects. Will adopt this for my gpdb and pg contributions.

**Comment by jocutajar** [comment]
> return `OPTIONS` to the docs

Both HEAD and OPTIONS are in the docs now.

> please calm down

You're right, and I'm calm. Just sharing feedback with you. If you appreciate free contributions, it's good to practice continuous integration to keep the contributors motivated.

**Comment by CyberDem0n** [comment]
👍

**Comment by hughcapet** [comment]
👍

**Comment by hughcapet** [comment]
Merged. Thank you, @dlax, for your much appreciated contribution and the highly valuable feedback.

**Pull request merged**